### DISCUSSION TITLE
DP aware breakpoints/media queries for Mobile Devices

### DISCUSSION THREAD

**Discussion by kasvith** [jpudysz/react-native-unistyles#395] [Feature Requests]
Hi Team,

This is a problem i came across while using uniStyles with various mobile device sizes.
In RN we uses DP to measure units, this is totally different from Web as in Web the pixel sizes and everything is handled by browsers 

But if we consider a Native mobile app things are really difficult to handle as some devices(apple) has DPR values going upto 3.0

So in uniStyles if we want to look for a smaller mobile screen like iPhoneSE

**Viewport Size:** 320px × 568px
**Screen Size:** 640px × 1136px
**Device Pixel Ratio:** DPR 2

the width returned by uniStyles is 640 for iPhoneSE, but actually the screen size is 320

So the breakpoints, media queries we define doesnt work really well since uniStyle doesnt seems care about the DPR and returning the screen size(View Port Size x DPR)

this is a really difficult situation as in web it works(our breakpoints are in Pixels like in tailwind/bootstrap but they hold true to the view port size instead of true pixel count in web), but in RN we should also consider the DPR to get sizes

So i belive uniStyles should either handle this internally or provide utilities to handle it

from developer's perspective i dont care about the DPR(as how it should be with RN)

**Comment by jpudysz** [comment]
Hey @kasvith we will discuss it internally and back to you

**Reply by kasvith** [reply]
hi @jpudysz anything about this?

**Reply by jpudysz** [reply]
Yes, I asked the team, and multiple ppl on Discord. Everyone is against it. Breakpoints are for web, tablets. I also never used them for mobile. 

I think you just need plugins like in v2. I will try to support them before RC.

**Reply by kasvith** [reply]
Well not really breakpoint, what im asking is unified width/height based on DPs instead of true pixel width...
Currently UniStyles uses true pixel width instead of viewport width in MQs etc

you have to divide it by Pixel Ratio to get actual width of screen

In web this is not a problem since we can set `initial-width=1.0` and all works seamlessly between breakpoints discarding the need of even thinking about DPs

**Reply by jpudysz** [reply]
So you want to upgrade `mq` to support values divided by pixel ratio (for mobile)?

**Reply by kasvith** [reply]
exactly...at least a version of it

so we can target smaller devices by just viewport dimensions with DP instead of worrying about true pixel width

since RN defaults to DP, MQs using true pixel width doesnt click in my head tbh

**Reply by kasvith** [reply]
Current setup is bit unnatural with when RN uses DP instead of true pixel dimensions...at least thats how i felt when using uniStyles

for web..current setup will just work as web doesnt care about this anyways

**Reply by jpudysz** [reply]
> since RN defaults to DP, MQs using true pixel width doesnt click in my head tbh

Now it makes sense. They should match breakpoints and use values divided by pixels. I totally agree (at least for mobile).

Sorry, that I didn't understand what's the problem

**Reply by kasvith** [reply]
This was driving me insane while working with mobile in uniStyles 

we discussed this in discord as well, glad that you got whats wrong...this is a very crucial thing to have

**Reply by kasvith** [reply]
if you can also release a patch for v2 as well

**Reply by kasvith** [reply]
hey @jpudysz any news about this?

**Reply by jpudysz** [reply]
It's planned for `beta.6`, `beta.5` required a lot of stability improvements. 

Also I have 2-3 tasks for `2.x` branch, so I will release it together. Most likely end of January 🙏

**Reply by jpudysz** [reply]
https://github.com/jpudysz/react-native-unistyles/issues/517

**Comment by jpudysz** [comment]
Implemented #541

**Comment by rappeldo** [comment]
Hi @jpudysz, thanks for implementing this feature for v3! Since our app is stuck on v2 for now, are you able to add this configuration for v2 as well?

**Reply by jpudysz** [reply]
Hey, what preventing you from update?

As of now I don't have any plans to release new features on v2 branch.

**Reply by rappeldo** [reply]
We are stuck with v2 since New Architecture is required for v3, we can't enable New Architecture for this time being due to some incompatibility issues.
